android/BOINC/app/src/main/java/edu/berkeley/boinc/SettingsFragment.kt

```diff
@@ -32,10 +33,17 @@ import edu.berkeley.boinc.utils.setAppTheme
 import kotlinx.coroutines.async
 import kotlinx.coroutines.coroutineScope
 import kotlinx.coroutines.launch
+import java.io.File
+import java.util.concurrent.ThreadLocalRandom
+import kotlin.streams.asSequence
 
 class SettingsFragment : PreferenceFragmentCompat(), SharedPreferences.OnSharedPreferenceChangeListener {
     private val hostInfo = BOINCActivity.monitor!!.hostInfo // Get the hostinfo from client via RPC
     private val prefs = BOINCActivity.monitor!!.prefs
+    private val charPool : List<Char> = ('a'..'z') + ('A'..'Z') + ('0'..'9')
```

Comment:
Could be constant

Reply:
In kotlin constant it mean static member to the class, then I need to do companion object. 
It more code and we don't gain performance out of that.

Reply:
val also mean that you cannot change the value. so it already constant.